PostFreeSurfer/PostFreeSurferPipelineLongLauncher.sh

```diff
@@ -0,0 +1,256 @@
+		PREP-TP) start_stage=0 ;;
+		PREP-T) start_stage=1 ;;
+		POSTFS-TP1) start_stage=2 ;;
+		POSTFS-T) start_stage=3 ;;
+		POSTFS-TP2) start_stage=4 ;;
+		*) log_Err_Abort "Unrecognized option for start-stage: $StartStage. Must be one of PREP-TP, PREP-T, POSTFS-TP1, POSTFS-T, POSTFS-TP2"
+	esac
+fi
+
+end_stage=4
+if [ -n "$EndStage" ]; then	
+	case $EndStage in
+		PREP-TP) end_stage=0 ;;
+		PREP-T) end_stage=1 ;;
+		POSTFS-TP1) end_stage=2 ;;
+		POSTFS-T) end_stage=3 ;;
+		POSTFS-TP2) end_stage=4 ;;
+		*) log_Err_Abort "Unrecognized option for end-stage: $EndStage. Must be one of PREP-TP, PREP-T, POSTFS-TP1, POSTFS-T, POSTFS-TP2"
+	esac
+fi
+
+#########################################################################################
+# Organising and cleaning up the folder structure
+#########################################################################################
+# ----------------------------------------------------------------------
+log_Msg "Organising and cleaning up the folder structure"
+# ----------------------------------------------------------------------
+
+LongDIR="${StudyFolder}/${Subject}.long.${LongitudinalTemplate}/T1w"
+for TP in ${Timepoints[@]} ; do
+  log_Msg "Organizing the folder structure for: ${TP}"
+  # create the symlink
+	TargetDIR="${StudyFolder}/${TP}.long.${LongitudinalTemplate}/T1w"
+	mkdir -p "${TargetDIR}"
+	ln -sf "${LongDIR}/${TP}.long.${LongitudinalTemplate}" "${TargetDIR}/${TP}.long.${LongitudinalTemplate}"
+
+  # remove the symlink in the subject's folder
+  rm -rf "${LongDIR}/${TP}"
+done
+
+##########################################################################################
+# PostFreesurferPipelineLongPrep.sh processing
+##########################################################################################
+#process timepoints
+if (( start_stage==0 )); then 
+	echo "################# PREP-TP Stage processing ########################"
+	for TP in ${Timepoints[@]}; do
+		echo "################# PREP-TP Stage processing ########################"
+	    echo "Running ppFS-long for timepoint: $TP"
+	    cmd=(${HCPPIPEDIR}/PostFreeSurfer/PostFreesurferPipelineLongPrep.sh \
+		--subject="$Subject" --path="$StudyFolder" \
+		--longitudinal-template="$LongitudinalTemplate"         \
+		--sessions="$TP"                         \
+		--template_processing=0                    \
+		--t1template="$T1wTemplate"                \
+		--t1templatebrain="$T1wTemplateBrain"      \
+		--t1template2mm="$T1wTemplate2mm"          \
+		--t2template="$T2wTemplate"                \
+		--t2templatebrain="$T2wTemplateBrain"      \
+		--t2template2mm="$T2wTemplate2mm"          \
+		--templatemask="$TemplateMask"             \
+		--template2mmmask="$Template2mmMask"       \
+		--fnirtconfig="$FNIRTConfig"               \
+		--freesurferlabels="$FreeSurferLabels"     \
+	    )	    
+	    par_add_job_to_stage $parallel_mode $fslsub_queue ${cmd[*]}
+	done
+	par_finalize_stage $parallel_mode $max_jobs
+fi
+
```

Comment:
```suggestion

if ((end_stage < 1)); then exit 0; fi

```
This would be less error-prone than needing to test both start and end in every conditional.  I don't really care for messages like "completed main functionality", and I wouldn't miss them as a result of early exit.

Reply:
ok

Reply:
ok